[NFC] Update function names in MCTargetAsmParser.h

lei137 opened the pull request:
Update function names to adhere to LLVM coding standard.

lei137 committed:
[NFC] Update function names in MCTargetAsmParser.h

Update function names to adhere to LLVM coding standard.

lei137 committed:
apply clang format

s-barannikov commented on llvm/include/llvm/MC/MCParser/MCTargetAsmParser.h:
```diff
@@ -439,14 +439,14 @@ class MCTargetAsmParser : public MCAsmParserExtension {
   ///        ownership of them to the caller.
   /// \return True on failure.
-  virtual bool ParseInstruction(ParseInstructionInfo &Info, StringRef Name,
+  virtual bool parseInstruction(ParseInstructionInfo &Info, StringRef Name,
```
The comment needs updating.

arsenm replied:
By just removing the reference to the function name, it's not necessary anymore

s-barannikov commented on llvm/include/llvm/MC/MCParser/MCTargetAsmParser.h:
```diff
@@ -439,14 +439,14 @@ class MCTargetAsmParser : public MCAsmParserExtension {
   }
 
-  /// ParseDirective - Parse a target specific assembler directive
+  /// parseDirectives - Parse a target specific assembler directive
```
Per coding standard, the comment shouldn't include function name.
```suggestion
  /// Parses a target specific assembler directive.
```

s-barannikov commented on llvm/include/llvm/MC/MCParser/MCTargetAsmParser.h:
```diff
@@ -457,7 +457,7 @@ class MCTargetAsmParser : public MCAsmParserExtension {
   ///
   /// \param DirectiveID - the identifier token of the directive.
-  virtual bool ParseDirective(AsmToken DirectiveID) { return true; }
+  virtual bool parseDirectives(AsmToken DirectiveID) { return true; }
```
It might be better to leave this method alone (it is deprecated).

lei137 committed:
Address review comments

1. remove function name from doc above functions being updated.
2. leave ParseDirective() as is as it's deprecated.

lei137 committed:
clean up doc that contain function names
